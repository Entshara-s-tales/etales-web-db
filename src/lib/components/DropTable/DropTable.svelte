<script lang="ts">
	import type { DropTable } from '$lib/db/units';
	import { slide } from 'svelte/transition';
	import { quintOut } from 'svelte/easing';

	export let dropTable: DropTable[];
</script>

<div
	class="mt-10 absolute left-0 top-0 shadow p-2"
	transition:slide={{ delay: 100, duration: 300, easing: quintOut }}
>
	{#each dropTable as itemDrop}
		<p>Item: {itemDrop.item.name}</p>
		<p>Type: {itemDrop.item.type}</p>
		<p>State: {itemDrop.item.state}</p>
		<p>DropRate: {itemDrop.dropRate * 100}</p>
	{/each}
</div>
