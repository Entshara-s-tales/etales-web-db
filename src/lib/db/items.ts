/**
 * Unknown right now, probably should be some kind of an enum
 * Seems that there's dress, head, unkown/not defined, weapon/wings, Miscelaneous
 */
export type ItemType = string;

// This isn't probably right though.
export type ItemState = 'passive';

export interface Item {
	name: string;
	type: ItemType;
	state: ItemState;
	description?: string;
}

export const MockItems: Item[] = [
	{
		name: 'Potato dress',
		type: 'Dress',
		state: 'passive',
		description: 'This item has a very calm aura',
	},
	{
		name: 'cat',
		type: 'Miscelaneous',
		state: 'passive',
	},
	{
		name: 'dog',
		type: 'Miscelaneous',
		state: 'passive',
		description: 'It definitely is not a cat!',
	},
];
