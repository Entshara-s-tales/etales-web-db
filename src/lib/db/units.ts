import { MockItems } from "./items";
import type { Item } from "./items";

export interface DropTable {
	item: Item;
	/**
	 * As percentages, normalized by 100 in frontend
	 */
	dropRate: number;
}

export interface Unit {
	name: string;
	id: number;
	// n <= 0 && n <= 99
	level: number;
	dropTable: DropTable[];
}

const MockDropTable = [
	{
		item: MockItems[0],
		dropRate: 0.1,
	},
	{
		item: MockItems[1],
		dropRate: 0.001,
	},
	{
		item: MockItems[2],
		dropRate: 0.8,
	},
];

export const MockUnits: Unit[] = [
	{
		name: "Spider pig",
		id: 0,
		level: 54,
		dropTable: MockDropTable,
	},
	{
		name: "Baby Bear",
		id: 64,
		level: 2,
		dropTable: MockDropTable,
	},
	{
		name: "Murloc Tiderunner",
		id: 93,
		level: 8,
		dropTable: MockDropTable,
	},
	{
		name: "Murloc Tiderunner Murloc Tiderunner",
		id: 93,
		level: 8,
		dropTable: MockDropTable,
	},
	{
		name: "Murloc Tiderunner Murloc Tiderunner Murloc Tiderunner",
		id: 93,
		level: 8,
		dropTable: MockDropTable,
	},
];
