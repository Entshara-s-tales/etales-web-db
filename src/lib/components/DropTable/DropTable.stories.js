import { MockUnits } from '$lib/db/units';
import DropTable from './DropTable.svelte';

export default {
  title: 'lib/components/DropTable',
  component: DropTable,
  argTypes: {},
};

const Template = ({ onClick, ...args }) => ({
  Component: DropTable,
  props: {
    ...args,
  },
});

export const SampleUsage = Template.bind({});
SampleUsage.args = {
  label: 'DropTable',
  unit: MockUnits[0],
};
