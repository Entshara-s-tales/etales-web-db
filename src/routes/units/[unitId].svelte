<script lang="ts">
	import { MockUnits } from '$lib/db/units';

	export let unitId: number;
	const unit = MockUnits.find((x) => x.id === unitId);
</script>

{#if unit}
	<div class="flex flex-row">
		<h3 class="flex-grow text-black">{unit.name}</h3>
		<img src="https://placekitten.com/120/120" alt="" />
	</div>
{/if}
