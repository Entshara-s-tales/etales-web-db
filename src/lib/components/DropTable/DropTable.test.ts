import { render } from '@testing-library/svelte';
import DropTable from './DropTable.svelte';

import { MockUnits } from '$lib/db/units';

describe('<DropTable/>', () => {
	it('matches snapshot', () => {
		const results = render(DropTable, {
			unit: MockUnits[0],
		});
		expect(results.container).toMatchSnapshot();
	});
});
