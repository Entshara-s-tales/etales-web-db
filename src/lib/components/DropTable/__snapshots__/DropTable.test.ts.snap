// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropTable/> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="w-full"
    >
      <div
        class="absolute right-0 top-0 "
      >
        <button
          class="text-black rounded"
          type="button"
        >
          <div
            class="w-10 h-10"
            inverse="false"
            pulse="false"
            spin="false"
          >
            <i
              classname="fa fa-times"
            />
          </div>
        </button>
      </div>
       
      <h3
        class="text-center pr-10 pl-2 pt-2 pb-2"
      >
        Spider pig
      </h3>
       
      <div
        class="mb-4 p-2 leading-snug"
      >
        <img
          alt=""
          class="m-auto pb-4"
          src="https://placekitten.com/80/80"
        />
         
        <p
          class=""
        >
          Name: 
          Potato dress
        </p>
         
        <p
          class=""
        >
          Type: 
          Dress
        </p>
         
        <p
          class=""
        >
          State: 
          Passive
        </p>
         
        <p
          class=""
        >
          Drop chance: 
          10
          %
        </p>
      </div>
       
      <hr
        class="w-10/12 p-4 m-auto border-gray-200"
      />
      
      <div
        class="mb-4 p-2 leading-snug"
      >
        <img
          alt=""
          class="m-auto pb-4"
          src="https://placekitten.com/80/80"
        />
         
        <p
          class=""
        >
          Name: 
          Cat
        </p>
         
        <p
          class=""
        >
          Type: 
          Miscelaneous
        </p>
         
        <p
          class=""
        >
          State: 
          Passive
        </p>
         
        <p
          class=""
        >
          Drop chance: 
          0.1
          %
        </p>
      </div>
       
      <hr
        class="w-10/12 p-4 m-auto border-gray-200"
      />
      
      <div
        class="mb-4 p-2 leading-snug"
      >
        <img
          alt=""
          class="m-auto pb-4"
          src="https://placekitten.com/80/80"
        />
         
        <p
          class=""
        >
          Name: 
          Dog
        </p>
         
        <p
          class=""
        >
          Type: 
          Miscelaneous
        </p>
         
        <p
          class=""
        >
          State: 
          Passive
        </p>
         
        <p
          class=""
        >
          Drop chance: 
          80
          %
        </p>
      </div>
       
      
    </div>
  </div>
</body>
`;
